// tinysong/lib/tinysong.js
// Generated by CoffeeScript 1.6.2
(function() {
  var req, request;

  request = require('request');

  req = function(method, query, opts, cb) {
    if (opts == null) {
      opts = {};
    }
    if (!module.exports.API_KEY) {
      throw "API_KEY not set\ntinysong.API_KEY = 'your_api_key' # Get yours at http://tinysong.com/api";
      return cb('API_KEY not set');
    }
    return request.get({
      uri: "http://tinysong.com/" + method + "/" + query,
      json: true,
      qs: {
        format: 'json',
        key: module.exports.API_KEY,
        limit: opts.limit != null ? opts.limit : void 0
      }
    }, function(err, resp, json) {
      if (err || (json.error != null)) {
        return cb(err || json.error);
      } else {
        if ((method === 'a' || method === 'b') && Array.isArray(json)) {
          return cb(null, null);
        } else {
          return cb(null, json);
        }
      }
    });
  };

  module.exports = {
    API_KEY: '',
    a: function(query, cb) {
      return req('a', query, {}, function(err, res) {
        return cb(err, res);
      });
    },
    b: function(query, cb) {
      return req('b', query, {}, function(err, res) {
        return cb(err, res);
      });
    },
    s: function(query, limit, cb) {
      if (typeof limit === 'function') {
        cb = limit;
        limit = 5;
      }
      return req('s', query, {
        limit: limit
      }, function(err, res) {
        return cb(err, res);
      });
    }
  };

}).call(this);
